core/app/sorters/spree/products/sort.rb

```diff
@@ -0,0 +1,43 @@
+module Spree
+  module Products
+    class Sort
+      def initialize(scope, params)
+        @scope = scope
+
+        @desc_order = String(params[:sort])[0] == '-'
+        @sort_field = @desc_order ? params[:sort][1..-1] : params[:sort]
```

Comment:
Can we just assign the value from params here and move this logic to a separate methods / attribute readers?
This looks rather confusing and I'd prefer initializers to be really simple